nixos/modules/services/networking/prosody.nix

```
@@ -1003,12 +1003,14 @@ in
         {
           User = cfg.user;
           Group = cfg.group;
-          Type = "forking";
+          Type = "simple";
           RuntimeDirectory = [ "prosody" ];
           PIDFile = "/run/prosody/prosody.pid";
-          ExecStart = "${cfg.package}/bin/prosodyctl start";
+          ExecStart = "${cfg.package}/bin/prosody -F";
```

Perhaps set `meta.mainProgram = "prosidy"`? Then you may 
```suggestion
          ExecStart = "${lib.getExe cfg.package} -F";
```